homeassistant/components/vasttrafik/config_flow.py

```diff
@@ -0,0 +1,384 @@
+"""Config flow for Västtrafik integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any, TypedDict
+
+import vasttrafik
+import voluptuous as vol
+
+from homeassistant.config_entries import (
+    ConfigEntry,
+    ConfigFlow,
+    ConfigFlowResult,
+    ConfigSubentryData,
+    ConfigSubentryFlow,
+    SubentryFlowResult,
+)
+from homeassistant.const import CONF_DELAY, CONF_NAME
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.selector import (
+    SelectOptionDict,
+    SelectSelector,
+    SelectSelectorConfig,
+    TextSelector,
+    TextSelectorConfig,
+)
+
+from .const import (
+    CONF_DEPARTURES,
+    CONF_FROM,
+    CONF_HEADING,
+    CONF_KEY,
+    CONF_LINES,
+    CONF_SECRET,
+    CONF_STATION_GID,
+    CONF_STATION_NAME,
+    CONF_TRACKS,
+    DEFAULT_DELAY,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_KEY): str,
```

Comment:
```suggestion
        vol.Required(CONF_KEY): TextSelector(),
```
Use the selectors on all fields (it was mentioned in some previous comment from emontnemery)